eva/readers/decord_reader.py

```diff
@@ -56,10 +62,23 @@ def __init__(
         self._predicate = predicate
         self._sampling_rate = sampling_rate or 1
         self._sampling_type = sampling_type
+        self._read_audio = read_audio
+        self._read_video = read_video
         super().__init__(*args, **kwargs)
 
     def _read(self) -> Iterator[Dict]:
         decord = _lazy_import_decord()
+        if self._read_audio:
+            try:
+                if "audio" in [
+                    stream["codec_type"]
+                    for stream in ffmpeg.probe(self.file_url)["streams"]
```

Comment:
I think we can remove the ffmpeg dependency by catching 'DECORDError' which is thrown by AVReader when a file with only video stream is read